catalystneuro/nwb-conversion-tools
# Update ecephys write for BaseRecording

## Motivation

This is a simple PR to update the ecephys write functions to work on `BaseRecording`'s, including `RecordingExtractors` wrapped by the API map from SpikeInterface.

This may also be the better solution to automatically resolving @h-mayorquin gain issue with interface that have multiple `SubRecording` nestings.

## How to test the behavior?
Tests just need to be debugged, maybe add some new ones to verify new functionality.

## Checklist
